File: src/BaseHmiTypes/Screens/Controls/HmiAuditTrailFieldPresentation.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAuditTrailFieldPresentation
{
    public HmiAuditTrailField Field { get; set; }

    public string? SourceType { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<double>? Width { get; set; }

    public string? TimeAndDateFormat { get; set; }

    public HmiMultilingualText? HeaderText { get; set; }
}

File: typescript/base-hmi-types/src/screens/controls/HmiAuditTrailFieldPresentation.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base.js";
import { HmiAuditTrailField } from "./HmiAuditTrailField.js";

export class HmiAuditTrailFieldPresentation {
  field = HmiAuditTrailField.Unknown;
  sourceType?: string;
  visible?: HmiProperty<boolean>;
  width?: HmiProperty<number>;
  timeAndDateFormat?: string;
  headerText?: HmiMultilingualText;
}
